src/services/external/excalidraw-backend/excalidraw.redis.server.factory.provider.ts

```diff
@@ -0,0 +1,322 @@
+      adapter: redisAdapter,
+    });
+
+    wsServer.on(CONNECTION, async socket => {
+      const agentInfo = await getUserInfo(
+        kratosClient,
+        socket.handshake.headers,
+        logger,
+        authService
+      );
+
+      if (!agentInfo) {
+        closeConnection(socket, 'Not able to authenticate user');
+        return;
+      }
+
+      logger?.verbose?.(
+        `User '${agentInfo.userID}' established connection`,
+        LogContext.EXCALIDRAW_SERVER
+      );
+
+      wsServer.to(socket.id).emit(INIT_ROOM);
+      // client events ONLY
+      socket.on(
+        JOIN_ROOM,
+        async (roomID: string) =>
+          await joinRoomEventHandler(
+            roomID,
+            agentInfo,
+            socket,
+            wsServer,
+            whiteboardRtService,
+            authorizationService,
+            logger
+          )
+      );
+
+      socket.on(SERVER_BROADCAST, (roomID: string, data: ArrayBuffer) =>
+        serverBroadcastEventHandler(roomID, data, socket)
+      );
+
+      socket.on(
+        SERVER_VOLATILE_BROADCAST,
+        (roomID: string, data: ArrayBuffer) =>
+          serverVolatileBroadcastEventHandler(roomID, data, socket)
+      );
+
+      socket.on(
+        DISCONNECTING,
+        async () =>
+          await disconnectingEventHandler(agentInfo, wsServer, socket, logger)
+      );
+
+      socket.on(DISCONNECT, () => disconnectEventHandler(socket));
+    });
+  } catch (error) {
+    logger.error(error, LogContext.EXCALIDRAW_SERVER);
+  }
+};
+
+/* Sets the user into the context field or closes the connection */
+const authenticate = async (
+  kratosFrontEndApi: FrontendApi,
+  headers: Record<string, string | string[] | undefined>,
+  logger: LoggerService,
+  authService: AuthenticationService
+): Promise<AgentInfo> => {
+  const cookie = headers.cookie as string;
+
+  try {
+    const { data: session } = await kratosFrontEndApi.toSession({
+      cookie,
+    });
+
+    if (!session) {
+      logger.verbose?.('No Ory Kratos session', LogContext.EXCALIDRAW_SERVER);
+      return authService.createAgentInfo();
+    }
+
+    const oryIdentity = session.identity as OryDefaultIdentitySchema;
+    return authService.createAgentInfo(oryIdentity);
+  } catch (e: any) {
+    throw new Error(e?.message);
+  }
+};
+/* returns the user agent info */
+const getUserInfo = async (
+  kratosFrontEndApi: FrontendApi,
+  headers: Record<string, string | string[] | undefined>,
+  logger: LoggerService,
+  authService: AuthenticationService
+) => {
+  try {
+    return await authenticate(kratosFrontEndApi, headers, logger, authService);
+  } catch (e) {
+    const err = e as Error;
+    logger.error(
+      `Error when trying to authenticate with excalidraw server: ${err.message}`,
+      err.stack,
+      LogContext.EXCALIDRAW_SERVER
+    );
+    return undefined;
+  }
+};
+/* This event is coming from the client; whenever they request to join a room */
+const joinRoomEventHandler = async (
```

Comment:
Put all eventHandlers in a separate file. That way you will keep your file <200 lines (generally considered best practice) and it will have a better readability and logical split.